src/engine/enginemaster.cpp

```diff
@@ -366,16 +406,80 @@ void EngineMaster::process(const int iBufferSize) {
     // Mix all the PFL enabled channels together.
     m_headphoneGain.setGain(chead_gain);
 
-    if (m_bRampingGain) {
-        ChannelMixer::mixChannelsRamping(
-                m_headphoneGain, &m_activeHeadphoneChannels,
-                &m_channelHeadphoneGainCache,
-                m_pHead, iBufferSize);
+    // Mixxx gets input signals delayed by the input latency,
+    // so to preserve the relative timing of the input with the master and headphones
+    // outputs, delay the master, booth, and headphones outputs by the input latency.
+    // This way, if the microphone user is on beat with the output they hear coming from
+    // the speakers/headphones, the input will remain on beat at the expense of
+    // adding some latency.
```

Comment:
> I do not see the "before", I see the delay and the then the write to the sidechain.

What do you mean by not seeing the "before"? That you cannot see the entire signal path for every mode in one place? That is why I have added comments. Refactoring so the TalkoverMixMode is only checked in one place would increase code duplication. I do not think using pointers to differently named internal buffers would help either. For example, if m_pMaster was a pointer to m_pMixerOut for example, it would be confusing to apply the master gain and balance to m_pMaster. Everything is at its place in the signal chain for specific reasons.